src/Raven.Server/Documents/TimeSeries/TimeSeriesValuesSegment.cs

```diff
@@ -747,6 +772,23 @@ private static void ThrowValuesOutOfRange(int numberOfValues)
         {
             throw new ArgumentOutOfRangeException("TimeSeriesValuesSegment can handle up to 32 values, but had: " + numberOfValues);
         }
+
+        public bool InvalidLastValue()
+        {
+            if (Version != SegmentVersion.V50000)
+                return false;
+
+            if (NumberOfLiveEntries == 0)
+                return false;
+
+            foreach (var value in SegmentValues.Span)
+            {
+                if (double.IsNaN(value.Last))
+                    return true;
+            }
```

Comment:
Why do we check all the items in the array?

Reply:
Eventually this check is used to decide whether we can use the `StatefulTimestampValueSpan` of the segment or not.

Reply:
can you add a comment explaining the reasoning